src/test/java/com/github/checkstyle/regression/internal/GitUtils.java

```diff
@@ -63,6 +67,7 @@ public static File addAnEmptyFileAndCommit(Repository repository, String fileNam
             throws IOException, GitAPIException {
         try (Git git = new Git(repository)) {
             final File file = new File(repository.getDirectory().getParent(), fileName);
+            file.getParentFile().mkdirs();
```

Comment:
I wrote a `GitUtils.addAnEmptyFileAndCommit(repository, "src/com/foo/c.java")`, and it would fail if there is no `src/com/foo` directory existing. I am not very sure if this implementation is appropriate, for I am not very familiar with Java NIO in the past.